java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -187,6 +187,10 @@ class ReferenceModeStore private constructor(
     override fun off(callbackToken: Int) {
```

Comment:
It seems you got a workaround, but if we need we could make this suspending I think. Since the servicestore also is using runBlocking.

Reply:
I change it back to suspend again as I found neither `stores.clear()` nor `DirectStore.close()` would actually clean up cache as there are too many references to a store, so the situation becomes more complicated.

In short, we must invoke `DirectStore.off()` to unregister a store's client first before closing and evicting it. We did miss calling `DirectStore.off()` for all stores in `BackingStore`.

Moreover by current implementation of `ReferenceModeStore`, even if the container store is already closed but the refmodestore is still able to accept messages from proxy, backing store and container store, see `receiveQueue`. So we must wait until both container store is closed and no messages are in flight are met before actually closing a store and evicting it from cache.

Reply:
Done fixing all aforementioned.